dali/kernels/common/join/tensor_join_gpu.h

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_COMMON_JOIN_TENSOR_JOIN_GPU_H_
+#define DALI_KERNELS_COMMON_JOIN_TENSOR_JOIN_GPU_H_
+
+#include "dali/kernels/kernel.h"
+#include "dali/kernels/common/type_erasure.h"
+#include "dali/kernels/common/join/tensor_join_gpu_impl.h"
+
+namespace dali {
+namespace kernels {
+
+/**
+ * @brief Joins (concatenates or stacks) batches of tensors
+ *
+ * This kernel takes a number of batches of tensors as the input and produces a single
+ * batch of tensors, where each tensor is a result of joining (concatenating or stacking)
+ * the respective tensors in the inputs.
+ *
+ * @tparam T        type of the tensor element. sizeof(T) must be 1,2,4,8 or 16
+ * @tparam new_axis if true, stacking is performed, otherwise - concatenation
+ *
+ */
+template <typename T, bool new_axis>
+class TensorJoinGPU : public tensor_join::TensorJoinImplGPU<type_of_size<sizeof(T)>, new_axis> {
+  using U = type_of_size<sizeof(T)>;
+  using InListU = InListGPU<U>;
+ public:
+  using Base = tensor_join::TensorJoinImplGPU<U, new_axis>;
+
+  /**
+   * @param ctx             kernel context, not used
+   * @param get_input_shape a function called with an input index; returns a reference to a shape
+   *                        of the input at given index
+   * @param num_inputs      number of joined tensors
+   * @param axis            The axis along which the tensors are concatenated or stacked;
+   *                        when `new_axis` is true, a new axis with the length equal to the number
+   *                        of inputs is inserted at this position.
+   *                        Valid range:
+   *                          * 0 to sample_dim when `new_axis` is true
+   *                          * 0 to sample_dim - 1 when `new_axis` is false
+   *                        where sample_dim is the dimensionality of the inputs.
+   *
+   * @remarks Inputs must have the same dimensionality.
+   * Respective tensors in the input must have the same shape (if new_axis == `true`) or can
+   * differ at index `axis` (if new_axis == `false`).
+   */
+  KernelRequirements Setup(KernelContext &ctx,
```

Comment:
```suggestion
  KernelRequirements Setup(KernelContext &/* not used */,
```

Reply:
No good solution there. I think that in some build environments unnamed parameter generated a warning (and then `-Werror` made it worse)...

Reply:
I think we have that in code here or there. But it just a suggestion.